tls/s2n_tls13_handshake.c

```diff
@@ -85,6 +84,70 @@ int s2n_tls13_compute_shared_secret(struct s2n_connection *conn, struct s2n_blob
     return 0;
 }
 
+int s2n_tls13_compute_pq_hybrid_shared_secret(struct s2n_connection *conn, struct s2n_blob *shared_secret) {
+    notnull_check(conn);
+    notnull_check(shared_secret);
+
+    /* conn->secure.server_ecc_evp_params should be set only during a classic/non-hybrid handshake */
+    eq_check(NULL, conn->secure.server_ecc_evp_params.negotiated_curve);
+    eq_check(NULL, conn->secure.server_ecc_evp_params.evp_pkey);
+
+    struct s2n_kem_group_params *server_kem_group_params = &conn->secure.server_kem_group_params;
+    notnull_check(server_kem_group_params);
+    struct s2n_ecc_evp_params *server_ecc_params = &server_kem_group_params->ecc_params;
+    notnull_check(server_ecc_params);
+
+    struct s2n_kem_group_params *client_kem_group_params = conn->secure.chosen_client_kem_group_params;
+    notnull_check(client_kem_group_params);
+    struct s2n_ecc_evp_params *client_ecc_params = &client_kem_group_params->ecc_params;
+    notnull_check(client_ecc_params);
+
+    DEFER_CLEANUP(struct s2n_blob ecdhe_shared_secret = { 0 }, s2n_blob_zeroize_free);
+
+    if (conn->mode == S2N_CLIENT) {
```

Comment:
* Re code sharing: Because we're using ECC keys from different places, the only real duplicated code is the 5 lines where we compute the ECC shared secret. I didn't think it was worth abstracting that out. Feel free to disagree (alternatively, I'm open to other suggestions for code re-use between the two functions).
* In `s2n_crypto.h` I added a pointer `struct s2n_kem_group_params *chosen_client_kem_group_params;` that will be assigned during the handshake to point to the correct client key so that we don't have to find it here. (I took the advice from here https://github.com/awslabs/s2n/blob/9f3f6fa1934adde7eca0ad0f22c0557994438eb7/tls/s2n_tls13_handshake.c#L65-L66)